src/Microsoft.AspNetCore.Razor/RazorTemplateEngine.cs

```diff
@@ -363,26 +368,48 @@ public GeneratorResults GenerateCode(TextReader input, CancellationToken? cancel
                 throw new ArgumentNullException(nameof(input));
             }
 
-            className = (className ?? Host.DefaultClassName) ?? DefaultClassName;
-            rootNamespace = (rootNamespace ?? Host.DefaultNamespace) ?? DefaultNamespace;
-
-            // Run the parser
-            var parser = CreateParser(sourceFileName);
-            Debug.Assert(parser != null);
-            var results = parser.Parse(input);
-
-            // Generate code
-            var chunkGenerator = CreateChunkGenerator(className, rootNamespace, sourceFileName);
-            chunkGenerator.DesignTimeMode = Host.DesignTimeMode;
-            chunkGenerator.Visit(results);
-
-            var codeGeneratorContext = new CodeGeneratorContext(chunkGenerator.Context, results.ErrorSink);
-            codeGeneratorContext.Checksum = checksum;
-            var codeGenerator = CreateCodeGenerator(codeGeneratorContext);
-            var codeGeneratorResult = codeGenerator.Generate();
-
-            // Collect results and return
-            return new GeneratorResults(results, codeGeneratorResult, codeGeneratorContext.ChunkTreeBuilder.Root);
+            try
+            {
+                className = (className ?? Host.DefaultClassName) ?? DefaultClassName;
+                rootNamespace = (rootNamespace ?? Host.DefaultNamespace) ?? DefaultNamespace;
+
+                // Run the parser
+                var parser = CreateParser(sourceFileName);
+                Debug.Assert(parser != null);
+                var results = parser.Parse(input);
+
+                // Generate code
+                var chunkGenerator = CreateChunkGenerator(className, rootNamespace, sourceFileName);
+                chunkGenerator.DesignTimeMode = Host.DesignTimeMode;
+                chunkGenerator.Visit(results);
+
+                var codeGeneratorContext = new CodeGeneratorContext(chunkGenerator.Context, results.ErrorSink);
+                codeGeneratorContext.Checksum = checksum;
+                var codeGenerator = CreateCodeGenerator(codeGeneratorContext);
+                var codeGeneratorResult = codeGenerator.Generate();
+
+                // Collect results and return
+                return new GeneratorResults(results, codeGeneratorResult, codeGeneratorContext.ChunkTreeBuilder.Root);
+            }
+            catch (Exception ex)
+            {
+                var errorSink = new ErrorSink();
+                errorSink.OnError(
+                    SourceLocation.Undefined,
+                    RazorResources.FormatFatalException(sourceFileName, Environment.NewLine, ex.Message),
```

Comment:
The costumer could. However, the issue that this is solving is to prevent VS from crashing when bits of Razor are replaced via extensions. 

Is there an alternative you'd prefer?